R/cube-collapse-dimensions.R

```diff
@@ -0,0 +1,127 @@
+#' @rdname cube-computing
+#' @export
+setMethod("collapse.dimensions", "CrunchCube", function (x, margin=NULL) {
+    # ensure that we have sensible margin input
+    selecteds <- is.selectedDimension(x@dims)
+    check_margins(margin, selecteds)
+    
+    # translate from user-cube margins to real-cube margins and establish the
+    # two groups of margins: those to collapse and those to keep
+    margins_to_collapse <- user2real(margin, cube = x)
+    margins_to_keep <- setdiff(seq_along(x@dims), margins_to_collapse)
+    
+    # if there are any _items in the margin to collapse, be wary! Since there
+    # are (likely) multiple items in the array, each row of the dataset would
+    # have more than on item here. If we simply summed across them (like we do
+    # with other measures below) then we would inflate the number of responses.
+    collapsed_items <- endsWith(getDimTypes(x)[margins_to_collapse], "_items")
+    
+    # grab the old cube structure and over-write it with new subset data.
+    out <- x
+    
+    # iterate through measures, collapsing the dimension(s) specified.
+    out@arrays[] <- lapply(
+        out@arrays,
+        collapse_with, 
```

Comment:
This only works with measures that are counts. It's probably not worth implementing for mean et al. (and you probably can't for median et al.) since this function only exists to support `index.table`, and even then only to avoid another API request.

Reply:
Yeah, that was my plan. I'll add an error if a non-count measured cube is passed so that we don't even attempt this for other measures. We could decide later if we wanted to actually resort to querying for the other cube if someone ever wants it. But for now an error is probably just fine.